packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -890,7 +890,9 @@ export class DataModel<
             const columnScopes = new Set(def.scopes);
             const columnScope = first(def.scopes);
             const columnSource = sources.get(columnScope) as unknown[];
-            const column = columnSource.map((valueDatum, datumIndex) => {
+            const column = new Array<unknown>(columnSource.length);
```

Comment:
Use `.map` here - if you use `new Array(length)`, it basically forces the array into a `Map<number, element>`. That was one of the big wins for the high performance work

Reply:
Based upon https://jsbench.me/t6m85tuhgv/1, our best option seems to be to just use `new Array()` without `.map()` - the main issue is likely that the closure can't be easily optimised at runtime for this particular case.